Collector/ProfileClient.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+use Http\Client\Common\EmulatedHttpAsyncClient;
+use Http\Client\Common\EmulatedHttpClient;
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+use Psr\Http\Message\RequestInterface;
+
+class ProfileClient implements HttpClient, HttpAsyncClient
+{
+    /**
+     * @var HttpClient|HttpAsyncClient $client
+     */
+    private $client;
+
+    /**
+     * @var Collector
+     */
+    private $collector;
+
+    /**
+     * @param HttpClient|HttpAsyncClient $client
+     * @param Collector                  $collector
+     */
+    public function __construct($client, Collector $collector)
+    {
+        if ($client instanceof HttpAsyncClient && $client instanceof HttpClient) {
```

Comment:
Should I use it directly without testing anything or should I use a condition to juste use `FlexibleHttpClient` when only one of both interface is implemented?